src/test/util/setup_common.cpp

```diff
@@ -361,8 +361,10 @@ void TestChainSetup::mineBlocks(int num_blocks)
         m_coinbase_txns.push_back(b.vtx[0]);
     }
 
-    g_txindex = std::make_unique<TxIndex>(1 << 20, true);
-    assert(g_txindex->Start(m_node.chainman->ActiveChainstate()));
+    if (!g_txindex) {
```

Comment:
should there be a critical section? let's put it under :cs_main. That's test, so, I think it's not a performance issue

g_txindex is not atomic, `mineBlocks` in theory can be called from different threads

Reply:
This commit was a test commit to try and debug crash cause as the conditions to trigger the crash weren't figured out yet (they have been now, will document in PR description when ready). Will request review when more appropriate solution is pushed.

Reply:
>  don't think there should be a cs_main lock as TxIndex and BaseIndex makes those locks internally. 

the internal logic may be indeed have locks.
Though, check + assigning `g_txindex` is not atomic operation and requires a lock